### PR TITLE
Fix: negative CCF coordinates cause incorrect allen ID mapping

### LINKED ISSUE #28: CCFMorphologyMapper incorrectly annotates negative CCF coordinates due to NumPy negative indexing
## Summary
Hi,

I believe `CCFMorphologyMapper.annotate_morphology` has a bounds-checking bug for negative CCF voxel coordinates.

When transformed SWC points fall anterior to the CCF border, their CCF `x` coordinate can become negative. These points should be treated as outside the CCF annotation volume. Instead, some are being assigned posterior/cerebellar Allen IDs.

## Observed behavior

In transformed SWC JSON outputs, points with negative `x` coordinates are assigned cerebellar labels.

Example observed output:

```text
x = -311.421824
allenId = 951
structure = Pyramus (VIII)
structure path = /997/8/512/528/645/951/

### PR DESCRIPTION
**Pull request by carshadi**
Fixes #28 

This pull request improves the robustness of the CCF morphology annotation logic and enhances test coverage for edge cases. The main change is a fix to ensure that out-of-bounds or invalid compartment positions do not result in errors or unintended region assignments. Additionally, new unit tests are introduced to verify this behavior.

**Bug fix and robustness improvements:**

* Updated `annotate_morphology` in `ccf_annotation.py` to explicitly check for out-of-bounds and invalid voxel indices before annotating compartments, assigning a region ID of 0 when necessary instead of relying on exception handling.

**Testing improvements:**

* Added helper classes `_FakeStructureTree` and `_FakeReferenceSpaceCache` to facilitate mocking of structure lookup and reference space behavior in tests.
* Introduced a new test `test_annotate_morphology_rejects_out_of_bounds_voxels` in `test_ccf_annotation.py` to ensure that compartments with out-of-bounds or invalid coordinates are not annotated with region information, improving coverage of edge cases.
* Added `numpy` import to the test file to support the new test cases.

### CODE REVIEW

File: src/aind_morphology_utils/ccf_annotation.py [hunk 1]
```diff
@@ -67,17 +67,23 @@ def annotate_morphology(
...
+                or np.any(pos_px < 0)
+                or np.any(pos_px >= volume_shape)
+            ):
                 # ID of 0 indicates point is outside CCF space
                 # This can happen for e.g., poorly registered tracings, or for
                 # neurons that project to the spinal cord,
                 # or other areas not represented in the CCF space.
                 ccf_region_id = 0
+            else:
+                pos_px = pos_px.astype(int)
```
Comment (mathewsummers) [inline src/aind_morphology_utils/ccf_annotation.py]: I believe astype(int) conversion will always round down; rounding first may be slightly more accurate.

Comment (carshadi) [reply src/aind_morphology_utils/ccf_annotation.py]: `astype(int)` will truncate the floating point part, pushing both positive and negative values closer to 0. 

We could round before the cast, but that would be a behavioral change. Consider the following:

image

The current behavior (left) assumes that we want to use the voxel containing the (floating) point coordinate. Rounding would change it so that it uses the nearest voxel center, and could push points near the volume edge out of bounds.

If rounding, we should add an additional check like
```python
pos_px = np.dot(pos, inv_dir_mat)

if not np.all(np.isfinite(pos_px)):
    ccf_region_id = 0
else:
    pos_px = np.rint(pos_px).astype(int)
    if np.any(pos_px < 0) or np.any(pos_px >= volume_shape):
        ccf_region_id = 0
    else:
        ccf_region_id = self.volume[tuple(pos_px)]
```

I guess it depends on the scientific intent - whether we consider the voxel containing the point to be the target or the nearest neighbor. What do you think?

Comment (mathewsummers) [reply src/aind_morphology_utils/ccf_annotation.py]: Yes I agree that this would be a behavior change; I hadn't realized the behavior was a floor function prior to this PR. Given that we expect some error in the registration process, to me rounding seems more accurate. Is your concern that points near the edges of the brain will round to outside the brain? I would imagine only a small number of points would suddenly shift to being outside the annotation volume itself, but all other points would be marginally more accurate. But also I'm willing to drop it in favor of proceeding with the original intent of this PR fix.

Comment (carshadi) [reply src/aind_morphology_utils/ccf_annotation.py]: @mathewsummers I changed it to use the rounding behavior, I have no preference either way since I'm not doing the analysis on the reconstructions :)

**Review by mathewsummers — APPROVED** [review]
